### ISSUE TITLE
Incorrect error message from SAM parsing

### ISSUE THREAD

**Issue by jkbonfield** [samtools/htslib#90]
```
$ ./samtools view test/dat/mpileup.1.sam
[E::sam_parse1] missing SAM header
[W::sam_read1] parse error at line 446
[main_samview] truncated file.
```

This is not correct. It does have a header, but this demo file (for some unknown reason) doesn't have the @SQ lines, only @RG.   Regardless of the odd nature of the input data, it should probably state that it is missing the SAM @SQ line instead as this initially confused me.

**Comment by jmarshall** [comment]
(Related to header API, #293).

**Comment by valeriuo** [comment]
Fixed by 6eacc77e3df6257a5c0426adf31782fd872cdc2f
